In [2]:
from pycoingecko import CoinGeckoAPI
import requests
from bs4 import BeautifulSoup as bs
import pandas as pd
import numpy as np
import requests
from bs4 import BeautifulSoup as bs

In [3]:
cg = CoinGeckoAPI()

In [4]:
cg.get_price(ids='bitcoin', vs_currencies='usd')

{'bitcoin': {'usd': 41832}}

In [5]:
df_coins = pd.DataFrame.from_dict(cg.get_coins_list(), orient='columns')[['id', 'name', 'symbol']]
df_coins

id                            name  \
0                              01coin                          01coin   
1            0-5x-long-algorand-token        0.5X Long Algorand Token   
2       0-5x-long-altcoin-index-token   0.5X Long Altcoin Index Token   
3      0-5x-long-ascendex-token-token  0.5X Long AscendEx Token Token   
4        0-5x-long-bitcoin-cash-token    0.5X Long Bitcoin Cash Token   
...                               ...                             ...   
13293                            zyro                            Zyro   
13294                           zyrri                           Zyrri   
13295                   zytara-dollar                   Zytara Dollar   
13296                             zyx                             ZYX   
13297                  zzz-finance-v2                  zzz.finance v2   

         symbol  
0           zoc  
1      algohalf  
2       althalf  
3       asdhalf  
4       bchhalf  
...         ...  
13293      zyro  
13294       zyr  
13295      zusd  
13296       zyx  
13297     zzzv2  

[13298 rows x 3 columns]

In [6]:
coin = df_coins['id']
URL1 = "https://coinmarketcap.com/currencies/"
URL2 = "/holders/"
URL_final = URL1 + coin + URL2 
links = pd.DataFrame(URL_final)

In [7]:
for i in range(1000):
    a = requests.get(links.iat[i, 0])
    b = a.status_code
    df_coins.at[i, 'Status'] = b

df_coins.head(8)

id                            name    symbol  \
0                          01coin                          01coin       zoc   
1        0-5x-long-algorand-token        0.5X Long Algorand Token  algohalf   
2   0-5x-long-altcoin-index-token   0.5X Long Altcoin Index Token   althalf   
3  0-5x-long-ascendex-token-token  0.5X Long AscendEx Token Token   asdhalf   
4    0-5x-long-bitcoin-cash-token    0.5X Long Bitcoin Cash Token   bchhalf   
5         0-5x-long-bitcoin-token         0.5X Long Bitcoin Token      half   
6         0-5x-long-cardano-token         0.5X Long Cardano Token   adahalf   
7       0-5x-long-chainlink-token       0.5X Long Chainlink Token  linkhalf   

   Status  
0   200.0  
1   200.0  
2   200.0  
3   404.0  
4   200.0  
5   200.0  
6   200.0  
7   200.0

200 - получен ответ от сервера (монета нашлась)
404 - страница не найдена
403 - превышен лимит запросов

In [13]:
links.iat[0, 0]

'https://coinmarketcap.com/currencies/01coin/holders/'

In [22]:
link_p = r"https://coinmarketcap.com/currencies/fantom/holders/"
request = requests.get(links.iat[0, 0])
result = request.status_code
print(link_p, request, result)

https://coinmarketcap.com/currencies/fantom/holders/ <Response [200]> 200
